### PR TITLE
Adds banned UI tag component and theme

### LINKED ISSUE #2853: UI - add `Banned` tag
First you need to add a new them to the Tag component & then add the `Banned` version that includes the emoji.

Figma spec:

https://www.figma.com/file/fw2pDIUlkhCkJE79ZyTbjp/CoinMachine?node-id=3195%3A36646

The story behind:

#2711

### PR DESCRIPTION
**Pull request by agustif**
## Description
Create a new BannedTag component similar to Tag and uses the new emoji-goblin svg.
It also handles special letter-spacing and different paddings/margins

In order to review this, please add this code wherever you want on the UI for the Tag to show up.
```tsx
 <BannedTag  />
```
**New stuff** ✨
* added new css style themeBanned:
* uses pink color for text, and mod pink, 20% transparent (The 20% opacity on the background, makes this tag blend with the bg color under it, so it will look as needed on white components, but not otherwise?

# TODOs
 
- [x] Create new BannedTag component in core
- [x] adds new emoji as svg, for the emoji-goblin
- [x] has correct margins / paddings  and font styles

**Screenshot** 📸 

image

Resolves #2853

### CODE REVIEW

**Review by ArmandoGraterol — CHANGES_REQUESTED** [review]
From what I see on my machine, the icon is not aligned with the text:
FireShot Capture 484 - Colony - localhost

Also, the padding _may_ not be the one in the specs but it's what we have for the other tags so I guess it's good to go in that department

File: src/img/icons/emoji-goblin.svg [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<svg width="12" height="12" viewBox="0 0 12 12" fill="none" xmlns="http://www.w3.org/2000/svg">
```
[the remark this answers is not in the document]
Comment (chinins) [reply src/img/icons/emoji-goblin.svg:1]: Agree with Armando, it needs to be `<svg viewBox="0 0 30 30">`

File: src/modules/core/components/BannedTag/BannedTag.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import Tag from '~core/Tag';
+
+import styles from './BannedTag.css';
+
+const displayName = 'BannedTag';
+
+interface Props extends HTMLAttributes<HTMLSpanElement> {
+  /** Text to display in the tag */
+  text?: MessageDescriptor | string;
+  /** Text values for intl interpolation */
```
[the remark this answers is not in the document]
Comment (chinins) [reply src/modules/core/components/BannedTag/BannedTag.tsx:14]: Agree, both of the comments are not needed

**Review by chinins — CHANGES_REQUESTED** [review]
On top on Armando's comments, it's not aligned on my machine either. Plus you need to add an `index.ts` to the `BannedTag` to export the component.

Screenshot 2021-11-15 at 21 43 42

File: src/modules/core/components/BannedTag/BannedTag.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (agustif) [reply src/modules/core/components/BannedTag/BannedTag.tsx]: Done

**Comment by agustif** [comment]
Both changes addressed, also changed some other stuff to get the alignment right! Plase re-review thx!

**Review by chinins — CHANGES_REQUESTED** [review]
the tag doesn't look like it should:
Screenshot 2021-11-16 at 21 40 05

Something went wrong with the rebase or you didn't run provision at some point or something else, or you committed the changes that you shouldn't have (please see below). This should not be in this PR:
Screenshot 2021-11-16 at 21 40 42

There are linting errors in css.d.ts file (you probably didn't wait for its refresh or something else), this needs to be fixed:
Screenshot 2021-11-16 at 21 40 59

**Comment by agustif** [comment]
image

It's looking like this to me, can you share which resolution are you using to try? Thx

File: src/modules/core/components/BannedTag/BannedTag.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+import styles from './BannedTag.css';
+
+const displayName = 'BannedTag';
+
+interface Props extends HTMLAttributes<HTMLSpanElement> {
+  text?: MessageDescriptor | string;
+}
+
+const BannedTag = ({ text = 'Banned' }: Props) => {
```
Comment (chinins) [inline src/modules/core/components/BannedTag/BannedTag.tsx:15]: this shouldn't be just a string `Banned`, it should be created via react-intl. Please, check in other components how we do it and apply it here

**Comment by chinins** [comment]
1350 px width (13 inch macbook), i don't think the problem is in resolution. you can see another tag next to it for comparison.

Screenshot 2021-11-17 at 09 51 15

File: src/modules/core/components/BannedTag/BannedTag.tsx (hunk 3 above)
Comment (rdig) [reply src/modules/core/components/BannedTag/BannedTag.tsx:15]: I've already added a global id for this.

So you can just use `text={{ id: 'label.banned' }}`

File: src/img/icons.json [hunk 4]
```diff
@@ -105,4 +106,4 @@
     "Sanjeet Shrimp",
     "Norman Naked Mole Rat"
   ]
-}
+}
```
Comment (rdig) [inline src/img/icons.json:109]: Please add an EOF new line

Same as it was before. This is most likely removed automatically by your editor, so you'll have to tweak your settings to allow them

File: src/modules/core/components/Tag/Tag.tsx [hunk 5]
```diff
@@ -14,6 +14,7 @@ export interface Appearance {
     | 'danger'
     | 'pink'
     | 'blue'
+    | 'banned'
```
Comment (rdig) [inline src/modules/core/components/Tag/Tag.tsx:17]: What's the point of this, if you've added a separate `core` component for the banned tag in the end ?

File: src/modules/core/components/BannedTag/BannedTag.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  return (
+    <Tag appearance={{ fontSize: 'small', theme: 'banned', margin: 'none' }}>
+      <Icon
+        title="banned"
```
Comment (rdig) [inline src/modules/core/components/BannedTag/BannedTag.tsx:19]: Please use the message descriptor, as mentioned above

File: src/modules/core/components/BannedTag/BannedTag.tsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        name="emoji-goblin"
+        appearance={{ size: 'normal' }}
+        className={styles.icon}
+      />
+      <div className={styles.textSyles}>{text}</div>
```
Comment (rdig) [inline src/modules/core/components/BannedTag/BannedTag.tsx:24]: Please use `<FormattedMessage />` here in order to pass in the message descriptor id

**Review by rdig — CHANGES_REQUESTED** [review]
There's a lot of changes needed here, please note both my comments and Olga's.

Also, please remove the `colonyServer` and `subgraph` since obviously you did no changes in there. _(if you think you need help with that let me know)_

Please try to finish this today, as we're in a bit of a deadline.

File: src/modules/core/components/Tag/Tag.tsx (hunk 5 above)
Comment (agustif) [reply src/modules/core/components/Tag/Tag.tsx:17]: I'm wrapping Tag with the banned theme and the goblin emoji on the new component

Would you prefer me to not use Tag here and move the banned theme to the BannedTag component css's instead?

```
const BannedTag = ({ children, text, textValues, ...rest}: Props) => {
  return (
    <Tag appearance={{ fontSize: 'small', theme: 'banned', margin: 'none' }}>
      <Icon
        title="banned"
        name="emoji-goblin"
        appearance={{ size: 'normal' }}
        className={styles.icon}
      />
      <div className={styles.textSyles}>
              {text ? (
        <>
          {typeof text === 'string' ? (
            text
          ) : (
            <FormattedMessage {...text} values={textValues} />
          )}
        </>
      ) : (
        children
      )}
      </div>
    </Tag>
  );
};
```

**Comment by agustif** [comment]
Fixed the wrong margins, was testing in a wrong place which the parent make the tag behave differently, correctly esting on events list now looks like this:

image

File: src/modules/core/components/BannedTag/BannedTag.tsx (hunk 7 above)
Comment (agustif) [reply src/modules/core/components/BannedTag/BannedTag.tsx]: Addressed in new commits

File: src/modules/core/components/BannedTag/BannedTag.tsx (hunk 6 above)
Comment (agustif) [reply src/modules/core/components/BannedTag/BannedTag.tsx]: Addressed in new commits

File: src/img/icons.json (hunk 4 above)
Comment (agustif) [reply src/img/icons.json]: Done, indeed needed to change some settings in VSCode for add final newline

File: src/modules/core/components/BannedTag/BannedTag.tsx (hunk 3 above)
Comment (agustif) [reply src/modules/core/components/BannedTag/BannedTag.tsx]: Done, 
`                      <BannedTag text={{ id: 'label.banned' }} />
`

File: src/modules/core/components/Tag/Tag.tsx (hunk 5 above)
Comment (agustif) [reply src/modules/core/components/Tag/Tag.tsx]: Moved all code to new component, no Tag modifications

File: src/modules/core/components/BannedTag/BannedTag.css [hunk 8]
```diff
@@ -0,0 +1,37 @@
+.icon {
+  display: inline-block;
+  margin-top: -4px;
+  margin-right: 4px;
+  margin-left: -4px;
```
Comment (rdig) [inline src/modules/core/components/BannedTag/BannedTag.css:5]: Please combine this into the shorthand version:

```suggestion
  margin: -4px 4px -4px 0;
```

File: src/modules/core/components/BannedTag/BannedTag.tsx [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+import Icon from '~core/Icon';
+
+import styles from './BannedTag.css';
+
+const displayName = 'BannedTag';
+
+interface Props extends HTMLAttributes<HTMLSpanElement> {
+  text?: MessageDescriptor | string;
+  textValues?: { [key: string]: string };
```
Comment (rdig) [inline src/modules/core/components/BannedTag/BannedTag.tsx:12]: We have a defined type for this already, it's called `SimpleMessageValues`. Please use it:
```
import { SimpleMessageValues } from '~types/index';
```

File: src/modules/core/components/BannedTag/BannedTag.tsx [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+const BannedTag = ({ children, text, textValues, ...rest }: Props) => {
```
Comment (rdig) [inline src/modules/core/components/BannedTag/BannedTag.tsx:15]: Please add a default value to the `text` prop, the label I've told you about.

```suggestion
const BannedTag = ({
  children,
  text = { id: 'label.banned' },
  textValues,
  ...rest
}: Props) => {
```

**Review by rdig — DISMISSED** [review]
There's a couple more changes that are required before this is ready.

Please add them in then let me know so I can approve and you can merge.

File: src/modules/core/components/BannedTag/BannedTag.css (hunk 8 above)
Comment (agustif) [reply src/modules/core/components/BannedTag/BannedTag.css]: done!

File: src/modules/core/components/BannedTag/BannedTag.tsx (hunk 10 above)
Comment (agustif) [reply src/modules/core/components/BannedTag/BannedTag.tsx]: Addressed in latest commit

File: src/modules/core/components/BannedTag/BannedTag.tsx [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+import { SimpleMessageValues } from '~types/index';
+import Icon from '~core/Icon';
+
+import styles from './BannedTag.css';
+
+const displayName = 'BannedTag';
+
+interface Props extends HTMLAttributes<HTMLSpanElement> {
+  text?: MessageDescriptor | string;
+  textValues?: SimpleMessageValues };
```
Comment (rdig) [inline src/modules/core/components/BannedTag/BannedTag.tsx:12]: There's a leftover bracket here. `{` Please remove it.

_(That's why the linter is complaining and your build is failing)_

Comment (chinins) [inline src/modules/core/components/BannedTag/BannedTag.tsx:12]: You don't need the ending curly brace here - that's why there is a linting error that github is showin
Screenshot 2021-11-17 at 18 05 01
g

File: src/modules/core/components/BannedTag/BannedTag.tsx [hunk 12]
```diff
@@ -0,0 +1,48 @@
+import React, { HTMLAttributes } from 'react';
+import { FormattedMessage, MessageDescriptor } from 'react-intl';
+import { SimpleMessageValues } from '~types/index';
```
Comment (chinins) [inline src/modules/core/components/BannedTag/BannedTag.tsx:3]: Usually we leave an empty line after outside imports & the dApp imports

File: src/modules/core/components/BannedTag/BannedTag.tsx [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+}: Props) => {
+  return (
+    <span className={styles.themeBanned} {...rest}>
+      <Icon
+        title={{ id: 'label.banned' }}
+        name="emoji-goblin"
+        appearance={{ size: 'normal' }}
+        className={styles.icon}
+      />
+      <div className={styles.textSyles} {...rest}>
```
Comment (chinins) [inline src/modules/core/components/BannedTag/BannedTag.tsx:29]: It's wrong to have div inside of a span element.
Anyway, why are you recreating the markup & styles of a tag? You can just use `Tag` component & add styles, text & emoji.

**Review by chinins — CHANGES_REQUESTED** [review]
Please, address my comment. Moreover, it still doesn't look right (the emoji is too low & the left padding is too large).
Your implementation:
Screenshot 2021-11-17 at 18 31 08
specs:
Screenshot 2021-11-17 at 18 33 53

**Comment by agustif** [comment]
image

Uploading last commit which should fix all the issues, also implemented now where there where TODO's annotated by Raul!

If you go to members list and ban yourself, you can test it without adding it explicitly somewhere now.

File: src/modules/core/components/BannedTag/BannedTag.tsx (hunk 11 above)
Comment (agustif) [reply src/modules/core/components/BannedTag/BannedTag.tsx:12]: Fixed!

Comment (agustif) [reply src/modules/core/components/BannedTag/BannedTag.tsx]: Thanks for the feedback, should look good now please re-review thx! gn

File: src/modules/core/components/BannedTag/BannedTag.tsx (hunk 13 above)
Comment (agustif) [reply src/modules/core/components/BannedTag/BannedTag.tsx]: I have removed it, let me know if it's OK now! Thank you olga

File: src/modules/core/components/BannedTag/BannedTag.tsx (hunk 12 above)
Comment (agustif) [reply src/modules/core/components/BannedTag/BannedTag.tsx:4]: Fixed now, thank you olga

**Review by chinins — APPROVED** [review]
As discussed, we'll merge it now and fix comment and membersInfoPopover later

**Pull request merged**